src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -113,4 +114,24 @@ public function testProcessInlinesWhenThereAreMultipleReferencesButFromTheSameDe
         $this->assertFalse($container->hasDefinition('b'));
         $this->assertFalse($container->hasDefinition('c'), 'Service C was not inlined.');
     }
+
+    public function testDecoratePrivateServiceDontThrowException()
+    {
+        $container = new ContainerBuilder();
+        $container->setResourceTracking(false);
+
+        $foo = $container->register('foo', '\stdClass');
+        $foo->setPublic(false);
+
+        $bar = $container->register('bar', '\stdClaqs');
+        $bar->setDecoratedService('foo');
+
+        $container->compile();
+
+        try {
+            $container->getAlias('foo');
```

Comment:
getting aliases after compilation does not make sense. ``getAlias`` is not a runtime usage of the container (it is not part of the ContainerInterface).
So this test does not cover a supported usage of the component.